server/src/main/java/org/opensearch/autotagging/Rule.java

```diff
@@ -87,6 +93,23 @@ public static Rule fromXContent(final XContentParser parser, FeatureType feature
         return Builder.fromXContent(parser, featureType).build();
     }
 
+    /**
+     * Parses a source string into a Rule object
```

Comment:
I think we should mention the format of the String in this case it will be rule document in the system index.